torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -1037,17 +1048,19 @@ def __init__(
     def make_launcher(self) -> LauncherType: ...
 
     def _gen_launcher_code(self, scope, def_args, runner_args) -> LauncherType:
-        grid = GridExpr.from_meta(self.inductor_meta, self.config)
-        # grid.prefix is usually empty, grid.x_grid is something like `-(xnumel//-1024)`
-        lines = [
-            f"def launcher({', '.join(def_args)}, stream):",
-            *[f"    {line}" for line in grid.prefix],
-            f"    grid_0 = {grid.x_grid}",
-            f"    grid_1 = {grid.y_grid}",
-            f"    grid_2 = {grid.z_grid}",
-            f"    runner({', '.join(runner_args)})",
-        ]
-        exec("\n".join(lines), scope)
+        if not hasattr(self, "launcher_code"):
```

Comment:
Storing the launcher code like this allows us to cache it in FXGraphCache without having to rerun this step over again